From c0cf944281e738b18e1114f59412b8c739e404c3 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Thu, 18 Oct 2018 10:38:59 +0800
Subject: [PATCH] arm64: dts: rockchip: add otg vbus regulator for rk1808 evb

Add otg vbus regulator for rk1808 evb, it can provide
5V power supply for otg host.

Change-Id: Ib535993d5446c7fbf58e58ab748e8d565f81447f
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk1808-evb.dtsi | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk1808-evb.dtsi b/arch/arm64/boot/dts/rockchip/rk1808-evb.dtsi
index 122f7d8601f5..4f520931db99 100644
--- a/arch/arm64/boot/dts/rockchip/rk1808-evb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk1808-evb.dtsi
@@ -180,6 +180,17 @@
 		reset-gpios = <&gpio4 RK_PC0 GPIO_ACTIVE_LOW>;
 	};
 
+	vcc_otg_vbus: otg-vbus-regulator {
+		compatible = "regulator-fixed";
+		gpio = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&otg_vbus_drv>;
+		regulator-name = "vcc_otg_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+	};
+
 	vcc5v0_sys: vcc5v0-sys {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc5v0_sys";
@@ -637,6 +648,7 @@
 
 &u2phy_otg {
 	status = "okay";
+	vbus-supply = <&vcc_otg_vbus>;
 };
 
 &usb_host0_ehci {
@@ -690,6 +702,12 @@
 		};
 	};
 
+	usb2 {
+		otg_vbus_drv: otg-vbus-drv {
+			rockchip,pins = <0 RK_PB7 0 &pcfg_pull_none>;
+		};
+	};
+
 	wireless-bluetooth {
 		uart4_rts_gpio: uart4-rts-gpio {
 			rockchip,pins = <4 RK_PB7 0 &pcfg_pull_none>;
-- 
2.35.3

